### PR TITLE
[cli] Fix error when `--serialize` flags are used with PTBs

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

PR #17431 introduced the `dry_run_or_execute_or_serialize` function and messed up the output for the `sui client ptb` when the `serialize-` flags were used. This fixes it and adds a test.

## Test plan 

Added a new test.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL:
- [ ] CLI: Fixed a bug when passing the `serialize-*` flag to the `sui client ptb`, which used to return an error after a recent change.
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui/src/client_commands.rs [hunk 1]
```diff
@@ -2641,10 +2641,12 @@ pub(crate) async fn dry_run_or_execute_or_serialize(
...
-    assert!(
-        !serialize_unsigned_transaction || !serialize_signed_transaction,
-        "Cannot specify both --serialize-unsigned-transaction and --serialize-signed-transaction"
-    );
+
+    if serialize_signed_transaction && serialize_unsigned_transaction {
+        bail!(
+            "Cannot specify both --serialize-unsigned-transaction and --serialize-signed-transaction."
+        );
+    }
```
Comment (amnn) [inline crates/sui/src/client_commands.rs:2649]: Note that there is `anyhow::ensure!` that does something similar to assert, but using anyhow:

```suggestion
    ensure!(
        !serialize_unsigned_transaction || !serialize_signed_transaction,
        "Cannot specify both --serialize-unsigned-transaction and --serialize-signed-transaction."
    );
```

File: crates/sui/src/client_ptb/ptb.rs [hunk 2]
```diff
@@ -160,17 +160,19 @@ impl PTB {
...
-
-        let transaction_response =
-            if let SuiClientCommandResult::TransactionBlock(response) = transaction_response {
-                response
-            } else {
-                anyhow::bail!("Internal error. Cannot run the PTB")
-            };
+        let transaction_response = match transaction_response {
+            SuiClientCommandResult::DryRun(response) => {
+                println!("{}", Pretty(&response));
```
Comment (amnn) [inline crates/sui/src/client_ptb/ptb.rs:165]: Could this part be handled by adding the logic to pretty-print `DryRun(response)` into the `Display` impl for `SuiClientCommandResult`?

**Review by amnn — APPROVED** [review]
Thanks for the fix!

File: crates/sui/src/client_ptb/ptb.rs (hunk 2 above)
Comment (stefan-mysten) [reply crates/sui/src/client_ptb/ptb.rs:165]: It's already there so easy fix.

File: crates/sui/src/client_commands.rs (hunk 1 above)
Comment (stefan-mysten) [reply crates/sui/src/client_commands.rs:2649]: Ah yes, forgot about ensure!.

**Pull request merged**